Coding/server/lib/FirmwareHeader/FirmwareHeader.cpp

```diff
@@ -0,0 +1,160 @@
+    m_hashAlgBuffer(),
+    m_signAlgBuffer(),
+    m_signatureBuffer()
+{
+}
+
+FirmwareHeader::~FirmwareHeader()
+{
+}
+
+void FirmwareHeader::deserialize(const uint8_t* dataChunk, const uint16_t chunkSize)
+{
+    const uint16_t idEndIdx = ID_SIZE_BYTE;
+    const uint16_t targetEndIdx = idEndIdx + TARGET_SIZE_BYTE;
+    const uint16_t versionEndIdx = targetEndIdx + VERSION_SIZE_BYTE;
+    const uint16_t hashAlgEndIdx = versionEndIdx + HASH_ALG_SIZE_BYTE;
+    const uint16_t signAlgEndIdx = hashAlgEndIdx + SIGN_ALG_SIZE_BYTE;
+    const uint16_t signatureEndIdx = signAlgEndIdx + SIGNATURE_SIZE_BYTE;
+
+    uint16_t readIdx = 0;
+    uint8_t byteBuffer = 0;
+
+    uint16_t oldReadIdx = m_readHeaderBytes;
+    for (uint16_t byteIdx = oldReadIdx; byteIdx < oldReadIdx + chunkSize; byteIdx++)
+    {
+        if (HEADER_SIZE_BYTE > byteIdx)
+        {
+            byteBuffer = dataChunk[readIdx];
+
+            /* Write read data into corresponding output buffer */
+            if (idEndIdx > byteIdx)
+            {
+                m_idBuffer[byteIdx] = byteBuffer;
+            }
+            else if ((idEndIdx <= byteIdx) && (targetEndIdx > byteIdx))
+            {
+                m_targetBuffer[(byteIdx - idEndIdx)] = byteBuffer;
+            }
+            else if (byteIdx == 32)
+            {
+                m_versionBuffer[0] = byteBuffer;
+            }
+            else if ((versionEndIdx <= byteIdx) && (hashAlgEndIdx > byteIdx))
+            {
+                m_hashAlgBuffer[(byteIdx - versionEndIdx)] = byteBuffer;
+            }
+            else if ((hashAlgEndIdx <= byteIdx) && (signAlgEndIdx > byteIdx))
+            {
+                m_signAlgBuffer[(byteIdx - hashAlgEndIdx)] = byteBuffer;
+            }
+            else if ((signAlgEndIdx <= byteIdx) && (signatureEndIdx > byteIdx))
+            {
+                m_signatureBuffer[(byteIdx - signAlgEndIdx)] = byteBuffer;
+            }
+
+            readIdx++;
+            m_readHeaderBytes++;
+
+
+#ifdef ACTIVATE_LOGGING
+            const uint8_t PRINT_BUFFER_SIZE = 128;
+            const uint8_t MULTIPLICATOR = 100;
+            char buffer[PRINT_BUFFER_SIZE];
+            uint8_t processedBytsPercent = static_cast<uint8_t>((m_readHeaderBytes / (float)HEADER_SIZE_BYTE) * MULTIPLICATOR);
+
+            /* Print the data into the output buffer */
+            sprintf(buffer, "Processed %d of %d header bytes (%d%%)", m_readHeaderBytes, HEADER_SIZE_BYTE, processedBytsPercent);
+            LOG_DEBUG(String(buffer));
+#endif
+        }
+        else
+        {
+            break;
+        }
+    }
+}
+void FirmwareHeader::reset()
+{
+    m_readHeaderBytes = 0;
+    memset(m_idBuffer, 0, sizeof(m_idBuffer));
+    memset(m_targetBuffer, 0, sizeof(m_idBuffer));
+    memset(m_versionBuffer, 0, sizeof(m_idBuffer));
+    memset(m_hashAlgBuffer, 0, sizeof(m_idBuffer));
+    memset(m_signAlgBuffer, 0, sizeof(m_idBuffer));
+    memset(m_signatureBuffer, 0, sizeof(m_idBuffer));
```

Comment:
Wrong sizes, except the first one. Copy&Paste Error

Reply:
Fixed with db2c24a